### PR TITLE
Add Emote preview

### PR DESCRIPTION
**Pull request by Crystalwarrior**
https://user-images.githubusercontent.com/3470436/171692940-6277bb65-c90a-4967-89b8-fff08c420fbe.mp4

Emote preview window can be resized, is affected by flip/pre settings, and you can also use the right-click dropdown to select specific emote types (preanim, talk (b), idle (a), segway (c))

### CODE REVIEW

File: src/aoemotepreview.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#include "aoemotepreview.h"
+
+AOEmotePreview::AOEmotePreview(QWidget *parent, AOApplication *p_ao_app) : QWidget(parent)
+{
+  ao_app = p_ao_app;
+
+  ui_viewport = new QWidget(this);
+//  ui_vp_background = new BackgroundLayer(ui_viewport, ao_app);
```
Comment (Salanto) [inline src/aoemotepreview.cpp:8]: Do we really need to keep those commented out lines around?

**Comment by stonedDiscord** [comment]
wait this opens an extra window?

**Comment by Salanto** [comment]
> wait this opens an extra window?

It opens an independent window from the main client to display a character inside it. That window is scalable and keeps its scale when switching between emotes.

**Comment by oldmud0** [comment]
This is failing the builds. Please fix.

**Review by Salanto — CHANGES_REQUESTED** [review]

**Comment by stonedDiscord** [comment]
are they supposed to be animated

**Review by stonedDiscord — APPROVED** [review]

**Review by Salanto — APPROVED** [review]

**Comment by oldmud0** [comment]
This looks cool. Maybe for next version we can put the size on the title bar so it doesn't obstruct the preview?

**Pull request merged**